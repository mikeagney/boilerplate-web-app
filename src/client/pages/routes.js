import React from 'react';
import { Redirect } from 'react-router-dom';
import Characters from './characters';
import Gauntlet from './gauntlet';

/**
 * An entry in a React route table.
 * @typedef {Object} Route
 * @property {string} Route.key
 *  A (locally) unique key for the route.
 * @property {string} Route.path
 *  The path to the route.
 * @property {boolean} Route.exact
 *  Whether the path match must be exact.
 * @property {((props:any)=>JSX.Element)=} Route.render
 * @property {React.Component=} Route.component
 */
const routes = [
  {
    key: 'root',
    path: '/',
    exact: true,
    render: () => <Redirect to="/characters" />,
  },
  {
    key: 'characters',
    path: '/characters',
    exact: true,
    component: Characters,
  },
  {
    key: 'gauntlet',
    path: '/gauntlet',
    exact: true,
    component: Gauntlet,
  },
];

/**
 * @type {()=>Route[]}
 */
export default () => routes;
