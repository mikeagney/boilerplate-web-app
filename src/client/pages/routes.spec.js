import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import routes from './routes';

describe('React app route table', () => {
  it('will return valid objects', () => {
    // Arrange
    // Act
    const result = routes();

    // Assert
    expect(result.length).toEqual(3);
    result.forEach((route) => {
      expect(route).toBeDefined();
      expect(route.key).toEqual(expect.any(String));
      expect(route.path).toEqual(expect.any(String));
      const RouteMatch = route.component || route.render || route.children;
      expect(RouteMatch).toBeDefined();
      const wrapper = shallow(<RouteMatch />);
      expect(toJson(wrapper)).toMatchSnapshot(route.key);
    });
  });
});
