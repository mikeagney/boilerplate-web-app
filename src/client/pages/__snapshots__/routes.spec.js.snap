// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React app route table getRoutes will return valid objects: characters 1`] = `
<div>
  <h2>
    Characters
  </h2>
</div>
`;

exports[`React app route table getRoutes will return valid objects: gauntlet 1`] = `
<div>
  <h2>
    Gauntlet
  </h2>
</div>
`;

exports[`React app route table getRoutes will return valid objects: root 1`] = `
<Redirect
  to="/characters"
/>
`;
